docs/howto/build-guides/platforms.rst

```diff
@@ -0,0 +1,169 @@
+
+Building on ``amd64`` or ``riscv64`` will produce one charm that runs on all
+architectures on Ubuntu 24.04 systems.
+
+
+Create a set of charms for multiple architectures
+-------------------------------------------------
+
+Charms may contain architecture-specific code. To build charms for specific
+architectures, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+       build-on: [amd64]
+       build-for: [amd64]
+     riscv64:
+       build-on: [riscv64]
+       build-for: [riscv64]
+
+Because the ``build-on`` and ``build-for`` entries are identical for each
+platform, the shorthand notation can be used instead:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+     riscv64:
+
+The results are the same with either snippet. Building on ``amd64`` will
+produce one charm that runs on ``amd64``. Building on ``riscv64`` will produce
+one charm that runs on ``riscv64``.
+
+
+Create a charm for a different architecture
+-------------------------------------------
+
+A charm may require cross-compilation to build. To create a charm for a
+different architecture, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     riscv64-cross:
+       build-on: [amd64]
+       build-for: [riscv64]
+
+Building on ``amd64`` will produce one charm that runs on ``riscv64``.
+
+Note that the charm developer must ensure the charm is compatible with the
+target architectures. By default, the `charm`_,
+:ref:`python <craft_parts_python_plugin>`, :ref:`poetry <craft_parts_poetry_plugin>`,
+and :ref:`uv <craft_parts_uv_plugin>` plugins will install wheels for python packages
+for the ``build-on`` architecture rather than the ``build-for``. For more information,
+see `craft-parts#974`_.
+
+Create a set of charms for multiple OS releases
+-----------------------------------------------
+
+A charm can only run on a single OS release or ``base``, using
+``charmcraft.yaml`` nomenclature. A ``charmcraft.yaml`` can use multi-base
+syntax to create a set of charms, each for a different base. To do this, the
+base is defined in each platform entry instead of being defined with the
+top-level ``base`` and ``build-base`` keywords.
+
+To build a charm for Ubuntu 22.04 and a charm for Ubuntu 24.04, use the
+following ``charmcraft.yaml`` snippet which uses :ref:`multi-base
+notation<reference-platforms-multi-base>`:
+
+.. code-block:: yaml
+
+   platforms:
+     ubuntu-22.04-amd64:
+       build-on: [ubuntu@22.04:amd64]
+       build-for: [ubuntu@22.04:amd64]
+     ubuntu-24.04-amd64:
+       build-on: [ubuntu@24.04:amd64]
+       build-for: [ubuntu@24.04:amd64]
+
+The ``build-on`` and ``build-for`` entries are identical for each platform, so
+the :ref:`multi-base shorthand notation
+<reference-platforms-multi-base-shorthand>` can be used:
+
+.. code-block:: yaml
+
+   platforms:
+     ubuntu@22.04:amd64:
```

Comment:
In the text we ought to point out the trailing colons for these entries, since they're easy to miss.

Reply:
I agree, but where in the text? We show the shorthand notation in several places. Do we want to highlight it after each sample? (Maybe best if people are coming to a specific section for reference, but repetitive if they're reading the entire document.) After only the first sample? (The opposite problem)

Your call, but for the moment I haven't made any changes to this.

Reply:
I've also added a description of this in the reference file.

Reply:
Let's leave it for now. This could be a case where having a shiny note somewhere high-up in the page would be valuable.